### PR TITLE
Remove test cases for using bitemporal_option_merge! of ActiveRecord:::Bitemporal::Callbacks

### PR DESCRIPTION
**Pull request by Dooor**
## Summary

I added test cases for using bitemporal_option_merge! of ActiveRecord:::Bitemporal::Callbacks in https://github.com/kufu/activerecord-bitemporal/pull/123 PR.

But specifications that change behavior depending on whether using NestedAttributes or using setter methods are not intended to support by activerecord-bitemporal gem.

It only needs to be tested that association's callbacks are executed when record is updated with NestedAttributes.

This PR organizes only for necessary tests

### CODE REVIEW

**Review by osyo-manga — APPROVED** [review]
OK, Good.
Thanks :)

File: spec/activerecord-bitemporal/callbacks_spec.rb [hunk 1]
```diff
@@ -111,61 +96,34 @@ class EmployeeWithBitemporalCallbacksLog < Employee
...
-        end
-
-        context 'job_title with option `ignore_bitemporal_callbacks: true`' do
-          before { employee.job_title.bitemporal_option_merge!(ignore_bitemporal_callbacks: true) }
-          it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
-          it { expect { subject }.not_to change { employee.job_title.log } }
-        end
+      context 'employee does not have changes and job_title has changes' do
+        let(:update_attributes) { { job_title_attributes: { id: employee.job_title.id, name: "COO" } } }
+        it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
```
Comment (wata727) [inline spec/activerecord-bitemporal/callbacks_spec.rb:125]: I'm surprised that the callback is invoked for an unchanged record. Is this the natural behavior of Active Record?

Comment (Dooor) [reply spec/activerecord-bitemporal/callbacks_spec.rb:125]: This behavior is the same behavior as ActiveRecord.
Even if ActiveRecord instance has no changes, update callbacks will be executed when save method is called.

Comment (wata727) [reply spec/activerecord-bitemporal/callbacks_spec.rb:125]: Oops, I misunderstood.

**Review by wata727 — APPROVED** [review]

**Pull request merged**